client/src/utils/parser/robotCodeToSsotParsing/robotCodeToSsotParsing.js

```diff
@@ -0,0 +1,369 @@
+            instructionBlocks[instructionBlocks.length - 1].rpaTask =
+                getRpaTask(currentLine, splitPlaceholder);
+            instructionBlocks[instructionBlocks.length - 1].paramArray =
+                getRpaParameters(currentLine, splitPlaceholder, instructionBlocks);
+
+            // TODO: implement parsing of outputVariables from RPAf code
+        }
+    })
+    return (errorWasThrown ? undefined : instructionBlocks);
+}
+
+/**
+ * @returns dummy startMarker as JSON => currently MARKERS aren't defined
+ * in our RPAf-Syntax and therefore there aren't implemented
+ */
+const buildStartMarker = () => ({
+    "successorIds": [],
+    "id": getEventId(),
+    "type": "MARKER",
+    "name": "START",
+    "predecessorIds": []
+})
+
+/**
+ * @param {Object} predecessor as an Object to get the predecessorId
+ * @returns dummy endMarker as JSON => currently MARKERS aren't defined
+ * in our RPAf-Syntax and therefore there aren't implemented
+ */
+const buildEndMarker = (predecessor) => ({
+    "successorIds": [],
+    "id": getEventId(),
+    "type": "MARKER",
+    "name": "END",
+    "predecessorIds": [predecessor ? predecessor.id : "MarkerElement"]
+})
+
+/**
+ * @description builds the attributeObject for a single element
+ * @param {Object} currentElement current instruction element
+ * @param {Object} singleElementFromTasksSection the parsed Object from the RPAf Code  
+ * @param {String} robotId the id of the current robot / ssot 
+ * @returns attributeObject for a single attribute
+ */
+const buildSingleAttributeObject = (currentElement, singleElementFromTasksSection, robotId) => {
+    let { rpaTask } = singleElementFromTasksSection;
+    if (!rpaTask) rpaTask = 'no Task defined'
+
+    return {
+        activityId: currentElement.id,
+        rpaApplication: singleElementFromTasksSection.rpaApplication,
+        rpaTask,
+        robotId
+    }
+}
+
+/**
+ * @description builds the parameterObject for a single element
+ * @param {Object} singleAtrributeObject the attribute Object of the current activity
+ * @param {Object} singleElementFromTasksSection the parsed Object from the RPAf Code   
+ * @param {Array} taskAndApplicationCombinations all combinations of applications and tasks
+ * @returns parameterObject for a single attribute
+ */
+const buildSingleParameterObject = (singleAtrributeObject, singleElementFromTasksSection, taskAndApplicationCombinations) => {
+    const { rpaApplication, activityId, rpaTask, robotId } = singleAtrributeObject;
+    const singleParamArray = singleElementFromTasksSection.paramArray;
+
+    const combinationObject = taskAndApplicationCombinations.filter((singleCombinationObject) =>
+        (singleCombinationObject.Application === rpaApplication && singleCombinationObject.Task === rpaTask)
+    )[0];
+
+
+    const parameterArray = combinationObject.inputVars.map((singleInputVariable, index) => {
+        const singleParameterObject = {
```

Comment:
Why not just create a copy of the object and touch only those fields that need changes?

Reply:
It wasn't quite as easy to fix as I had imagined, but in the end you're absolutely right. Great point; I implemented it!

Reply:
It wasn't quite as easy to fix as I had imagined, but in the end you're absolutely right. Great point; I implemented it!